src/botmsg.c

```diff
@@ -42,21 +42,24 @@ static char OBUF[1024];
  */
 void tandout_but EGG_VARARGS_DEF(int, arg1)
 {
-  int i, x;
+  int i, x, len;
   char *format;
   char s[601];
   va_list va;
 
   x = EGG_VARARGS_START(int, arg1, va);
   format = va_arg(va, char *);
 
-  egg_vsnprintf(s, 511, format, va);
+  len = egg_vsnprintf(s, 511, format, va);
   va_end(va);
-  s[sizeof(s) - 1] = 0;
+  if (len >= sizeof(s)) {
```

Comment:
we only write 511, so to detect truncation (\0 skipping), we need to check for 511